src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -161,7 +161,12 @@ public boolean canProvision(Label label) {
                         node -> {
                           logger.info("Agent {} is online", name);
                           return node;
-                        });
+                        })
+                    .exceptionally(e -> {
+                      logger.info("Agent {} failed to come online", name);
```

Comment:
@jeschkies this block is needed to provide a callback to kill the agent when the future completes with any exception